src/components/LinkModal/LinkModal.js

```diff
@@ -0,0 +1,253 @@
+import selectors from 'selectors';
+
+import './LinkModal.scss';
+
+const LinkModal = () => {
+  const [isDisabled, isOpen, totalPages, currentPage, tabSelected] = useSelector(state => [
+    selectors.isElementDisabled(state, 'linkModal'),
+    selectors.isElementOpen(state, 'linkModal'),
+    selectors.getTotalPages(state),
+    selectors.getCurrentPage(state),
+    selectors.getSelectedTab(state, 'linkModal'),
+  ]);
+  const [t] = useTranslation();
+  const dispatch = useDispatch();
+
+  const urlInput = React.createRef();
+  const pageNumberInput = React.createRef();
+
+  const [dropdownNumbers, setDropdownNumbers] = useState([]);
+  const [url, setURL] = useState('');
+  const [pageNumber, setPageNumber] = useState(1);
+
+  const closeModal = () => {
+    dispatch(actions.closeElement('linkModal'));
+    setURL('');
+    setPageNumber(1);
+    core.setToolMode(defaultTool);
+  };
+
+  const createLink = () => {
+    const linksResults = [];
+
+    const quads = core.getSelectedTextQuads();
+    const selectedAnnotations = core.getSelectedAnnotations();
+
+    const currPageQuads = quads[currentPage - 1];
+
+    if (currPageQuads) {
+      const selectedText = core.getSelectedText();
+      currPageQuads.forEach(quad => {
+        const link = new Annotations.Link();
+        link.PageNumber = currentPage;
+        link.StrokeColor = new Annotations.Color(0, 165, 228);
+        link.StrokeStyle = 'underline';
+        link.StrokeThickness = 2;
+        link.Author = core.getCurrentUser();
+        link.Subject = 'Link';
+        link.X = Math.min(quad.x1, quad.x3);
+        link.Y = Math.min(quad.y1, quad.y3);
+        link.Width = Math.abs(quad.x1 - quad.x3);
+        link.Height = Math.abs(quad.y1 - quad.y3);
+        linksResults.push(link);
+      });
+      createHighlightAnnot(linksResults, currPageQuads, selectedText);
+    }
+
+    if (selectedAnnotations && selectedAnnotations.length === 1) {
+      const link = new Annotations.Link();
+      link.PageNumber = currentPage;
+      link.StrokeColor = new Annotations.Color(0, 165, 228);
+      link.StrokeStyle = 'underline';
+      link.StrokeThickness = 2;
+      link.Author = core.getCurrentUser();
+      link.Subject = 'Link';
+      link.X = selectedAnnotations[0].X;
+      link.Y = selectedAnnotations[0].Y;
+      link.Width = selectedAnnotations[0].Width;
+      link.Height = selectedAnnotations[0].Height;
+      linksResults.push(link);
+      selectedAnnotations[0].associateLink([link.Id]);
+    }
+
+    return linksResults;
+  };
+
+  const createHighlightAnnot = async(linkAnnotArray, quads, text) => {
+    const linkAnnot = linkAnnotArray[0];
+    const highlight = new Annotations.TextHighlightAnnotation();
+    highlight.PageNumber = linkAnnot.PageNumber;
+    highlight.X = linkAnnot.X;
+    highlight.Y = linkAnnot.Y;
+    highlight.Width = linkAnnot.Width;
+    highlight.Height = linkAnnot.Height;
+    highlight.StrokeColor = new Annotations.Color(0, 0, 0, 0);
```

Comment:
Can you remind me why we want to programmatically create a highlight annotation here? I remember we discussed about this a bit but I don't remember... Sorry

Reply:
we wanted to have consistent behaviour for deleting annots, for example create a shape link it, delete shape and link goes away. Selecting text didnt have associated shape or highlight. We create invisible highlight.